mutiny-core/src/redshift.rs

```diff
@@ -280,6 +280,28 @@ impl RedshiftManager for NodeManager {
                 .clone()
         };
 
+        // find the channel reserve the introduction channel
+        let reserve = match rs.introduction_channel {
+            None => 0,
+            Some(chan) => sending_node
+                .channel_manager
+                .list_channels()
+                .iter()
+                .find_map(|c| {
+                    if c.funding_txo.map(|u| u.into_bitcoin_outpoint()) == Some(chan) {
+                        c.unspendable_punishment_reserve
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or(0),
+        };
+
+        // original utxo value - opening tx fee - channel reserve
+        let max_sats = rs.amount_sats - rs.fees_paid - reserve;
```

Comment:
Are these unsigned? May need to do saturating subtract since your unwraps set it to 0.  How possible is it for those 0's to trigger? Shouldn't we always have this number?

Reply:
ah yeah you're right, i was thinking about the `amount_sats` part coming through as `0` but you're right that wouldn't happen.